packages/rum/src/domain/record/record.spec.ts

```diff
@@ -334,6 +334,22 @@ describe('record', () => {
       expect(inputRecords.length).toBe(1)
     })
 
+    it('should record the scroll event inside a shadow root', () => {
+      const div = appendElement('<div></div>', createShadow()) as HTMLDivElement
+      startRecording()
+      expect(getEmittedRecords().length).toBe(recordsPerFullSnapshot())
+
+      div.dispatchEvent(createNewEvent('scroll', { target: div, composed: false }))
+
+      recordApi.flushMutations()
+      const innerMutationData = getLastIncrementalSnapshotData<BrowserMutationData>(
+        getEmittedRecords(),
+        IncrementalSource.Scroll
+      )
+
+      expect(innerMutationData).toBeDefined()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could we ensure that there is only the scroll on the inner element that is generated?

Reply:
✔️